x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/whitelist/InternalSqlScriptUtils.java

```diff
@@ -382,8 +382,8 @@ public static Integer dateDiff(String dateField, Object dateTime1, Object dateTi
         return (Integer) DateDiffProcessor.process(dateField, asDateTime(dateTime1), asDateTime(dateTime2) , ZoneId.of(tzId));
     }
 
-    public static ZonedDateTime dateTrunc(String truncateTo, Object dateTime, String tzId) {
-        return (ZonedDateTime) DateTruncProcessor.process(truncateTo, asDateTime(dateTime) , ZoneId.of(tzId));
+    public static Object dateTrunc(String truncateTo, Object dateTime, String tzId) {
+        return DateTruncProcessor.process(truncateTo, asDateTime(dateTime, true), ZoneId.of(tzId));
```

Comment:
Why did you add `true` here, it shouldn't be needed.

Reply:
Since dateTrunc works with Interval and ZonedDateTime, function `asDateTime` with one argument forces to cast (ZonedDateTime) and some integration tests fails. As you can see some functions which works with Interval and ZonedDateTime like 
    `Object add(Object left, Object right)`
    `Object div(Object left, Object right) etc.`
doesn't use `asDateTime` and checks 
`if (dateTime instanceof JodaCompatibleZonedDateTime) `
in the Processor.
I was not sure to use `true` or `false`. I was only sure not to use `asDateTime` function with one argument which cast to `ZonedDateTime`. It works with `true` or `false` but doesn't work one argument.